apps/meteor/app/api/server/lib/maybeMigrateLivechatRoom.ts

```
@@ -0,0 +1,26 @@
+import { isOmnichannelRoom, type IRoom } from '@rocket.chat/core-typings';
+import { Rooms } from '@rocket.chat/models';
+import type { FindOptions } from 'mongodb';
+
+import { migrateVisitorIfMissingContact } from '../../../livechat/server/lib/contacts/migrateVisitorIfMissingContact';
+
+export async function maybeMigrateLivechatRoom(room: IRoom | null, options: FindOptions<IRoom> = {}): Promise<IRoom | null> {
+	if (!room || !isOmnichannelRoom(room)) {
+		return room;
```

should throw an error here instead